spec/controllers/api/v2/medical_histories_controller_spec.rb

```diff
@@ -45,9 +45,9 @@ def create_record_list(n, options = {})
     let(:facility_in_same_group) { FactoryBot.create(:facility, facility_group: request_user.facility.facility_group) }
     let(:facility_in_another_group) { FactoryBot.create(:facility) }
 
-    let(:patient_in_request_facility) { FactoryBot.create(:patient, registration_facility: request_facility) }
-    let(:patient_in_same_group) { FactoryBot.create(:patient, registration_facility: facility_in_same_group) }
-    let(:patient_in_another_group) { FactoryBot.create(:patient, registration_facility: facility_in_another_group) }
+    let(:patient_in_request_facility) { FactoryBot.build(:patient, registration_facility: request_facility) }
+    let(:patient_in_same_group) { FactoryBot.build(:patient, registration_facility: facility_in_same_group) }
+    let(:patient_in_another_group) { FactoryBot.build(:patient, registration_facility: facility_in_another_group) }
```

Comment:
This is a build since the patient factory creates a `medical_history` by default now, the controller spec create one separately and so there are two.

Reply:
This seems like a problem with the way we're asserting things in the spec. Can we stop asserting counts and start asserting the presence of specific medical history IDs instead? 

`build` is much faster, and that's great, but I'm not sure we should use it to avoid this fragile situation.

Reply:
I think we should avoid touching this territory for the scope of this PR to not explore.

Reply:
This change touches the shared specs and changing them is quite tricky. I would like to avoid going down that road unless really necessary.

I think it's ok to keep the `build`, since it's just an artefact of how our factories work now. It _will_ create a medical history in the background, which we don't want in this case.

Reply:
Yup, that sounds sensible. 👍